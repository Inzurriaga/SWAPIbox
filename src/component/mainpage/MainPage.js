import React, { Component } from "react";
import Buttons from "../buttons/Buttons";
import { ReactComponent as Star } from "../../image/star.svg"
import { ReactComponent as Wars } from "../../image/wars.svg"
import ContainerDecoration from "../../image/containerdec.png"
import PeoplesContainer from "../peoplescontainer/PeoplesContainer"
import PlanetsContainer from "../planetscontainer/PlanetsContainer"
import VehicleContainer from "../vehiclescontainer/VehiclesContainer"

export default class Intro extends Component{
    constructor(){
        super();
        this.state = {
            containerDisplay: "",
            peoplesinfo: [],
            planetsinfo: [],
            vehicleinfo: []
        }
    }

    componentDidMount = () => {
        // local storage setstate here for the favorite
    }

    switchContainer = (container) => {
        this.setState({
            containerDisplay: container
        })
    }

    saveContainerInfo = (containerName, containerInfo) => {
        this.setState({
            [containerName]: containerInfo
        })
    }

    render = () => {
        const {containerDisplay, peoplesinfo, planetsinfo, vehicleinfo } = this.state
        const componentContainer = {
            peopleContainer: (<PeoplesContainer saveContainerInfo={this.saveContainerInfo} 
                                                peoplesinfo={peoplesinfo}/>),
            planetsContainer: (<PlanetsContainer saveContainerInfo={this.saveContainerInfo} 
                                                planetsinfo={planetsinfo}/>),
            vehiclesContainer: (<VehicleContainer saveContainerInfo={this.saveContainerInfo} 
                                                vehicleinfo={vehicleinfo}/>)
        }
        return(
            <section className="main-section">
                <header>
                    <Star className="star-title"/>
                    <h2>
                        <span>S</span>
                        <span>w</span>
                        <span>a</span>
                        <span>p</span>
                        <span>i</span>
                        <span>-</span>
                        <span>B</span>
                        <span>o</span>
                        <span>x</span>
                    </h2>
                    <Wars className="wars-title"/>
                </header>
                <Buttons switchContainer={this.switchContainer}/>
                <img className="top-container-dec" src={ContainerDecoration} alt="top-container"/>
                {componentContainer[containerDisplay]}
                <img className="bottom-container-dec"src={ContainerDecoration} alt="bottom-container"/>
            </section>
        )
    }
}