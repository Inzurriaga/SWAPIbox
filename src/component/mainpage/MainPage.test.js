import React from "react"
import { shallow } from "enzyme"
import MainPage from "./MainPage"

describe("main_page", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <MainPage />
        )
    })
    it("should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()
    });
    it("should have a default state", () => {
        expect(wrapper.state()).toEqual({
            containerDisplay: "",
            peoplesinfo: [],
            planetsinfo: [],
            vehicleinfo: []
        })
    })
    it("should change the state of containerDisplay when function is invoke", () => {
        expect(wrapper.state("containerDisplay")).toEqual("")
        wrapper.instance().switchContainer("hello")
        expect(wrapper.state("containerDisplay")).toEqual("hello")
    })
    it("should change the container info when function is invoke", () => {
        expect(wrapper.state("peoplesinfo")).toEqual([])
        wrapper.instance().saveContainerInfo("peoplesinfo", [1,2,3])
        expect(wrapper.state("peoplesinfo")).toEqual([1,2,3])
    })
})