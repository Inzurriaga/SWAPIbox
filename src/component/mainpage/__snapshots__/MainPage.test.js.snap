// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main_page should match snapshot 1`] = `
<section
  className="main-section"
>
  <header>
    <ForwardRef
      className="star-title"
    />
    <h2>
      <span>
        S
      </span>
      <span>
        w
      </span>
      <span>
        a
      </span>
      <span>
        p
      </span>
      <span>
        i
      </span>
      <span>
        -
      </span>
      <span>
        B
      </span>
      <span>
        o
      </span>
      <span>
        x
      </span>
    </h2>
    <ForwardRef
      className="wars-title"
    />
  </header>
  <Buttons
    switchContainer={[Function]}
  />
  <img
    alt="top-container"
    className="top-container-dec"
    src="containerdec.png"
  />
  <img
    alt="bottom-container"
    className="bottom-container-dec"
    src="containerdec.png"
  />
</section>
`;
